Update module

jcriadop opened the pull request:
- Multi-label order generator (filtered by date)
- Better update/upgrade module process
- Voucher discount compatibility fixed
- Tax policies fixed
- Minor updates and bug fixed
- Added link to configuration when module is not installed

jcriadop committed:
Update module

- Multi-label order generator (filtered by date)
- Better update/upgrade module process
- Voucher discount compatibility fixed
- Tax policies fixed
- Minor updates and bug fixed
- Added link to configuration when module is not installed

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+	}
+	public function getAllOrders($id = null, $reference = null, $date_start = null, $date_end = null, $name = null, $address = null, $postal_code = null, $city  = null, $state_name = null, $country = null)
+	{
+		$sql = 'SELECT o.reference,o.id_order,o.date_add,o.reference, a.firstname, a.lastname, a.address1, a.address2, a.postcode, a.city, cl.name as country, s.name as state FROM '._DB_PREFIX_.'orders o INNER JOIN '._DB_PREFIX_.'seur_order so ON so.id_order  = o.id_order  INNER JOIN '._DB_PREFIX_.'address a ON o.id_address_delivery = a.id_address INNER JOIN '._DB_PREFIX_.'country_lang cl ON cl.id_coun ...
```
Please escape `$this->context->language->id` with an `(int)` cast

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+		$sql = 'SELECT o.reference,o.id_order,o.date_add,o.reference, a.firstname, a.lastname, a.address1, a.address2, a.postcode, a.city, cl.name as country, s.name as state FROM '._DB_PREFIX_.'orders o INNER JOIN '._DB_PREFIX_.'seur_order so ON so.id_order  = o.id_order  INNER JOIN '._DB_PREFIX_.'address a ON o.id_address_delivery = a.id_address INNER JOIN '._DB_PREFIX_.'country_lang cl ON cl.id_coun ...
+		$where = array();
+		if ((int)$id > 0)
+			$where[] = ' o.id_order ='.$id.' ';
```
Please escape the data stored in $id

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' o.id_order ='.$id.' ';
+		
+		if (trim($reference) != '')
+			$where[] = ' o.reference like \''.$reference.'\' ';
```
Please escape the data stored in $reference

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+		
+		
+		if (trim($date_start) != '' && trim($date_end) != '' )
+			$where[] = ' o.date_add between \''.$date_start.'\' AND   \''.$date_end.'\' ';
```
Please escape the data stored in your dates

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' o.date_add between \''.$date_start.'\' AND   \''.$date_end.'\' ';
+			
+		if (trim($date_start) != '' && trim($date_end) == '' )
+			$where[] = ' o.date_add >= \''.$date_start.'\'  ';
```
Please escape the data stored in $date_start

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' o.date_add >= \''.$date_start.'\'  ';
+			
+		if (trim($date_start) == '' && trim($date_end) != '' )
+			$where[] = ' o.date_add <= \''.$date_start.'\'  ';
```
Please escape the data stored in $date_start

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' o.date_add <= \''.$date_start.'\'  ';
+			
+		if (trim($name) != ''  )
+			$where[] = ' ( a.firstname like = \'%'.$name.'%\'  OR a.lastname like = \'%'.$name.'%\'  )  ';
```
Please escape the data stored in $name

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' ( a.firstname like = \'%'.$name.'%\'  OR a.lastname like = \'%'.$name.'%\'  )  ';
+		
+		if (trim($address) != ''  )
+			$where[] = ' ( a.address1 like = \'%'.$address.'%\'  OR a.address2 like = \'%'.$address.'%\'  )  ';
```
Please escape the data stored in $address

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+		
+		
+		if (trim($postal_code) != ''  )
+			$where[] = ' a.post_code like = \'%'.$postal_code.'%\'   ';
```
Please escape the data stored in $postal_code

jcriadop replied:
Solved

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' a.post_code like = \'%'.$postal_code.'%\'   ';
+		
+		if (trim($city) != ''  )
+			$where[] = ' a.city like = \'%'.$city.'%\'   ';
```
Please escape the data stored in $city

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' a.city like = \'%'.$city.'%\'   ';
+		
+		if (trim($state_name) != ''  )
+			$where[] = ' a.state like = \'%'.$state_name.'%\'   ';
```
Please escape the data stored in $state_name

jcriadop replied:
Solved.

Quetzacoalt91 commented on AdminSeur.php:
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
+			$where[] = ' a.state like = \'%'.$state_name.'%\'   ';
+		
+		if (trim($country) != ''  )
+			$where[] = ' a.country like = \'%'.$country.'%\'   ';
```
Please escape the data stored in $country

jcriadop replied:
Solved.

Quetzacoalt91 commented:
Hi,

jQuery, jQuery UI, jQuery UI datepicker already exist on PrestaShop. Please use them instead of including them a second time in your module.
BTW, the validation will be really simpler if you do this, because the diff will be lighter and with the same functionalities.

Thanks

Quetzacoalt91 commented:
Hi,

Your controller are are made to load all the data needed by a template. You should not use it to create the HTML code.
For instance, please move all the HTML content found in *controllers/admin/AdminSeurController.php* in a tpl file. If another files are also concerned, please do the same work.

Thanks

jcriadop committed:
escaped vars.

escaped vars.

jcriadop committed:
jquery and html adapted

Jquery lib removed from module to use prestashop jquery libs. HTML moved
from controller to tpl.

jcriadop commented:
Hi, new commit solve this.

jcriadop commented:
Hi, new commit solve this.

Quetzacoalt91 commented:
Thanks a lot, the PRs is really easier to review. :smiley:

Quetzacoalt91 commented on install/seurcashondelivery/seurcashondelivery.php:
```diff
@@ -503,34 +546,40 @@ public function validateOrderFORWEBS_v5($id_cart, $id_order_state, $amount_paid,
 
 					$order->total_products = (float)$this->context->cart->getOrderTotal(false, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
 					$order->total_products_wt = (float)$this->context->cart->getOrderTotal(true, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
+					$order->total_products_wt = (float)$this->context->cart->getOrderTotal(true, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
```
Be careful, you duplicated the previous line

Quetzacoalt91 commented:
### ajax/getLabelAjax.php
* SQL queries are not escaped

### config.xml
* should be in english

### Localized config_en.xml & config_fr.php
* Should be deleted

### Unused translation files to remove
* en.php
* fr.php

### Are these template files used ?
* views/templates/admin/header.tpl
* views/templates/admin/backoffice_header.tpl

jcriadop committed:
Minor fix

- ajax/getLabelAjax.php
SQL queries escaped
- config.xml
translated to english
- Localized config_en.xml & config_fr.php Should be deleted
don't find files ??
- Unused translation files to remove en.php - fr.php
removed
- Remove not used admin tpl files
removed

jcriadop commented:
Hi, All solved on last commit, except "Localized config_en.xml & config_fr.php", I don't find these files.

Quetzacoalt91 commented:
Sorry, I was talking about config_es.xml !

jcriadop commented:
Do you want I remove config_es.xml ??

Quetzacoalt91 commented:
Yes please , we only keep config.xml

jcriadop commented:
Done, we removed.

Quetzacoalt91 commented:
Hi,

I'm still waiting for a new commit, did you forget to push ?

jcriadop commented:
Hi, I sent last commit 2 days ago with last reques:

https://github.com/jcriadop/seur/commit/de2eb13e6f8e9b9b4e3424a8829155fdd8d78cb4

I don't see any question pending for me.

Quetzacoalt91 commented:
But config_es.xml in https://github.com/PrestaShop/seur/pull/1/files is still here. That's why

jcriadop commented:
Ok, commit sent

jcriadop committed:
config_es.xml file removed

config_es.xml file removed

Quetzacoalt91 commented:
Thanks, the PR has just been merged and the changes should be available from tomorrow.

Best regards

Quetzacoalt91 commented on controllers/admin/AdminSeur15Controller.php:
```diff
@@ -29,50 +29,481 @@
+	{
+	parent::__construct();
+	$this->module = Module::getInstanceByName('seur');
+	}
```
This part is already executed by the parent class

jcriadop replied:
Line removed.

Quetzacoalt91 commented on install/seurcashondelivery/seurcashondelivery.php:
```diff
@@ -307,7 +306,36 @@ public function hookPaymentReturn($params)
+
+		if ((int)$order->id >0)
+		{
+			$sql = 'SELECT codfee FROM '._DB_PREFIX_.'seur_order WHERE id_order='.$order->id;
```
You cast the variable in your test, but not in the SQL request. Please add a `(int)` cast

jcriadop replied:
Cast added.

Quetzacoalt91 commented on install/seurcashondelivery/seurcashondelivery.php:
```diff
@@ -307,7 +306,36 @@ public function hookPaymentReturn($params)
+	{
+		if ((int)$order->id >0)
+		{
+			$sql = 'SELECT total_paid FROM '._DB_PREFIX_.'seur_order WHERE id_order='.$order->id;
```
You cast the variable in your test, but not in the SQL request. Please add a `(int)` cast

jcriadop replied:
Cast added.

Quetzacoalt91 commented on install/seurcashondelivery/seurcashondelivery.php:
```diff
@@ -307,7 +306,36 @@ public function hookPaymentReturn($params)
+	{
+		if ((int)$order->id >0)
+		{
+			$sql = 'UPDATE '._DB_PREFIX_.'seur_order SET total_paid = '.$total_paid.' WHERE id_order='.$order->id;
```
And here again, please add a `(int)` cast.

jcriadop replied:
Cast added.
